tests/integration/test_end_point.py

```diff
@@ -195,6 +180,73 @@ def test_health_database_ko(setup_database):
         assert_get_health_200()
 
 
+def assert_health(expected_result):
+    if expected_result == 200:
+        assert_get_health_200()
+    elif expected_result == 503:
+        assert_get_health_with_ko_and_503()
+
+
+def get_health_data_provider():
+    services = list(product([True, False], repeat=4))
+    data_provider = []
+    for service_list in services:
+        data_provider.append(
+            {
+                "expected_result": 200 if all(s is True for s in service_list) else 503,
+                "database": service_list[0],
+                "output_rabbitmq": service_list[1],
+                "navitia": service_list[2],
+                "redis": service_list[3],
+            }
+        )
+
+    return data_provider
+
+
+@pytest.mark.parametrize("input", get_health_data_provider())
+def test_health_with_data_provider(input):
+    db_config = kirin.app.config[str("SQLALCHEMY_DATABASE_URI")]
+    expected_result = input.get("expected_result")
+    database_status = input.get("database")
+    output_rabbitmq_status = input.get("output_rabbitmq")
+    navitia_status = input.get("navitia")
+    redis_status = input.get("redis")
+
+    class EmptyMocker:
+        def __init__(self, *args, **kwargs):
+            pass
+
+        def __enter__(self, *args, **kwargs):
+            pass
+
+        def __exit__(self, *args, **kwargs):
+            pass
+
+    redis_mock = RedisTemporaryMocker if redis_status is False else EmptyMocker
+    rabbit_mock = RabbitmqTemporaryMocker if output_rabbitmq_status is False else EmptyMocker
+
+    with requests_mock.mock() as m, redis_mock("wrong_redis_url") as _red, rabbit_mock(
+        "wrong_url", "wrong_exchange", "wrong_exchange_type"
+    ) as rab:
```

Comment:
minor, as you don't use it
```suggestion
    ) as _rab:
```